### PR TITLE
Fix working with UNC paths and reparse points on Windows

### PR DESCRIPTION
**Pull request by belonesox**
# What does this PR do?

Fix working with UNC paths (Samba folders / Shared VM folders / etc)
https://github.com/Nuitka/Nuitka/issues/2767

# Why was it initiated? Any relevant Issues?
https://github.com/Nuitka/Nuitka/issues/2767

Now it works on windows both with UNC paths mapped to folder and even unmapped UNC paths (`\\somecomp\somefolder`). 

# PR Checklist

- [x] Correct base branch selected? Should be `develop` branch.
- [x] Enabled commit hook or executed `./bin/autoformat-nuitka-source`.
- [V] All tests still pass. 
   - On windows tests only on Python 3.12 (several hours)
   - On Linux failed on Python 2, but looks python 2 broken on my box

I think no documentation changes needed.
About tests, theoretically in possible checking working UNC paths on Windows, I can try to write it if needed, but not sure that possible to cover nontrivial cases (Windows Sandox, Hyper-V shared folders, etc).

### CODE REVIEW

File: nuitka/utils/FileOperations.py [hunk 1]
```diff
@@ -236,9 +236,14 @@ def _getRealPathWindows(path):
             if str is not bytes:
                 result = result.decode("utf8")
 
-            _real_path_windows_cache[path] = os.path.join(
-                os.path.dirname(path), result.rstrip("\r\n")
-            )
+            if result.startswith("UNC"):
+                # Avoid network mounts being converted to UNC shared paths by newer
+                # Python versions, many tools won't work with those.
+                _real_path_windows_cache[path] = path
```
[the remark this answers is not in the document]
Comment (kayhayen) [reply nuitka/utils/FileOperations.py:242]: I agree with the bot here, this probably needs to involve a backslash of some sorts.

[the remark this answers is not in the document]
Comment (belonesox) [reply nuitka/utils/FileOperations.py]: Add one slash for reliability, retested on mapped and unmapped drives.

File: nuitka/utils/FileOperations.py [hunk 2]
```diff
@@ -288,35 +293,34 @@ def getFilenameRealPath(path):
...
+    # Avoid network mounts being converted to UNC shared paths by newer
     # Python versions, many tools won't work with those.
-    if os.name == "nt" and path.startswith("\\\\"):
-        drive, _remaining_path = os.path.splitdrive(orig_path)
-
-        if drive:
-            drive_real_path = os.path.realpath(drive + "\\")
-            assert path.startswith(drive_real_path)
+    if os.name == "nt":
+        if path.startswith("\\\\"):
```
[the remark this answers is not in the document]
Comment (belonesox) [reply nuitka/utils/FileOperations.py:299]: Disagree with bot here. Here we isolate Windows case ('os.name == "nt"') with two different branches (UNC and non-UNC inside).

[the remark this answers is not in the document]
Comment (kayhayen) [reply nuitka/utils/FileOperations.py:299]: I agree with you, combining is not always more readable

**Comment by kayhayen** [comment]
So, this is a bit ugly, as it became too big of a code, I will move it out to a function after merge.

**Pull request merged**